jmix-search/search/src/main/java/io/jmix/search/utils/FileParserResolverManager.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.search.utils;
+
+import io.jmix.core.FileRef;
+import io.jmix.search.exception.EmptyFileParserResolversList;
+import io.jmix.search.exception.UnsupportedFileTypeException;
+import io.jmix.search.index.fileparsing.FileParserResolver;
+import org.apache.tika.parser.Parser;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The service that searches appropriate file parsers for the supported file types.
+ * A search principle is based on the sequential applying FileParserResolver objects' checks for the given file.
+ */
+@Component("search_FileParserResolverManager")
+public class FileParserResolverManager {
+
+    protected List<FileParserResolver> fileParserResolvers;
+
+    public FileParserResolverManager(List<FileParserResolver> fileParserResolvers) {
+        this.fileParserResolvers = fileParserResolvers;
+    }
+
+    public Parser getParser(FileRef fileRef) throws UnsupportedFileTypeException {
+        if (fileParserResolvers.isEmpty()) {
+            throw new EmptyFileParserResolversList();
+        }
+
+        String fileName = fileRef.getFileName();
+
+        List<String> messages = new ArrayList<>();
+
+        for (FileParserResolver resolver : fileParserResolvers) {
+            if (resolver.supports(fileRef)) {
+                return resolver.getParser();
+            }
+            messages.add(resolver.getCriteriaDescription());
```

Comment:
If we try to process completely unsupported file it will collect "criteria descriptions" from all existing resolvers.
As a result the final exception message will contain multiple "The file extension should be one of the following: ..." expressions.

That is the point in favor of the `FileParserResolver#getCriteriaDescription` method irrelevance.

This loop should collect extensions supporter by unsuitable resolvers and provide them to exception if non of existing resolvers is suitable. And exception generates the final message based on provided extensions.

Reply:
Discussed. Left without changes.